crates/rpc/src/handlers/state.rs

```diff
@@ -1,10 +1,35 @@
+use alloy_primitives::FixedBytes;
 use ream_consensus::deneb::beacon_state::BeaconState;
-use ream_storage::{db::ReamDB, tables::Table};
+use ream_storage::{db::ReamDB, tables::{ Field, Table}};
+use warp::{
+    http::status::StatusCode, 
+    reject::Rejection, 
+    reply::{with_status, Reply}
+};
 
 use crate::types::{errors::ApiError, id::ID};
 
+use super::BeaconResponse;
+
 pub async fn get_state_from_id(state_id: ID, db: &ReamDB) -> Result<BeaconState, ApiError> {
     let block_root = match state_id {
+        ID::Named(ref name) => match name.as_str() {
+            "head" => {Ok(Some(FixedBytes::new([0; 32])))},
+            "justified" => {
+                let justified_checkpoint = db.justified_checkpoint_provider().get().map_err(|_| ApiError::InternalError)?
+                .ok_or_else(|| ApiError::NotFound(String::from("Justified checkpoint not found")))?;
+                
+                Ok(Some(justified_checkpoint.root))
+            },
+            "finalized" => {
+                let finalized_checkpoint = db.finalized_checkpoint_provider().get().map_err(|_| ApiError::InternalError)?
+                .ok_or_else(|| ApiError::NotFound(String::from("Finalized checkpoint not found")))?;
+                
+                Ok(Some(finalized_checkpoint.root))
+            },
+            "genesis" => db.slot_index_provider().get(0),
+            &_ => Ok(Some(FixedBytes::new([0; 32])))
+        },
```

Comment:
```suggestion
        ID::Head | ID::Genesis => return Err(ApiError::NotFound(format!(
        "This ID type is currently not supported: {state_id:?}"
    ))),
    ID:Justified => {
          let justified_checkpoint = db.justified_checkpoint_provider().get().map_err(|_| ApiError::InternalError)?
          .ok_or_else(|| ApiError::NotFound(String::from("Justified checkpoint not found")))?;
          
          Ok(Some(justified_checkpoint.root))
      },
    ID:Finalized => {
            let finalized_checkpoint = db.finalized_checkpoint_provider().get().map_err(|_| ApiError::InternalError)?
            .ok_or_else(|| ApiError::NotFound(String::from("Finalized checkpoint not found")))?;
            
            Ok(Some(finalized_checkpoint.root))
      },
```

I am not sure if we will end up supporting the genesis state, or we don't do currently so we should just throw an error